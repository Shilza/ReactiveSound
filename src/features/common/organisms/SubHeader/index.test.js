import React from 'react';
import {render} from 'react-testing-library';
import {SubHeader} from "./index";

describe('Test SubHeader', () => {
    const location = 'Favorite';
    const children = <div data-testid='subHeaderChildren'>Children</div>;

    const {container, getByTestId, queryByTestId} = render(
        <SubHeader section={<div>Section</div>} location={location}>
            {children}
        </SubHeader>
    );

    it('should renders correctly', () => {
        expect(container.firstChild).toMatchSnapshot();
    });

    it('title should have correct text', () => {
        const title = getByTestId('subHeaderTitle');
        expect(title).toHaveTextContent(location);
    });

    it('section should have two child', () => {
        let section = queryByTestId('subHeaderSection');
        expect(section.childNodes.length).toBe(2);
    });

    it('should have passed child', () => {
        const subHeaderChildren = queryByTestId('subHeaderChildren');
        expect(container).toContainElement(subHeaderChildren);
    });
});