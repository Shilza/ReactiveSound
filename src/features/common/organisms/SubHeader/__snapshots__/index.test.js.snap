// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SubHeader should renders correctly 1`] = `
<div
  class="subHeader"
>
  <div
    class="container"
  >
    <div
      class="section"
      data-testid="subHeaderSection"
    >
      <div>
        Section
      </div>
      <span>
         /
      </span>
    </div>
    <h1
      class="title"
      data-testid="subHeaderTitle"
    >
      Favorite
    </h1>
    <div
      data-testid="subHeaderChildren"
    >
      Children
    </div>
  </div>
</div>
`;
